R/util.R

```diff
@@ -184,6 +184,78 @@ util.as_numeric_if_number <- function(x){
     return(x)
 }
 
+util.is_vector_of_logicals <- function(x){
+  # Are all elements of x logicals (permitting blanks, NOT permitting vectors entirely 
+  # composed of numeric booleans, i.e., 0/1)?
+  # (regardless of whether x is numeric)
+  # Character vectors sometimes need to be changed to logicals,
+  # e.g., when all columns set to character types by default.
+  
+  # if all the elements of x are numbers, return FALSE. This is to ensure correct handling of
+  # binary 0/1 variables as numbers, not logicals
+  if(util.is_vector_of_numbers(x)){
+    return(FALSE)
+  }
+  
+  # find logical values
+  x_as_logical <- suppressWarnings(as.logical(x))
+  
+  # anything that gets coerced to NA by as.numeric is not really a number.
+  # but elements that were blank originally are ok.
+  non_logical <- is.na(x_as_logical)
+  originally_blank <- util.is_blank(x)
+  
+  # Return FALSE if there are any coerced blanks, because that means x 
+  # contained elements that could not be converted to numeric.
+  # Otherwise return TRUE.
+  if(any(non_logical & !originally_blank)){
+    return(FALSE)
+  }
+  else{
+    return(TRUE)
+  }
+}
+
+util.is_vector_of_logicals_test <- function(){
+  x_char <- c("TRUE", "FALSE", "a", "b")
+  x_logical <- c("TRUE", "FALSE")
+  x_0_1_only <- c(1, 0)
```

Comment:
nit: I'd expect a broad range of numbers to also pass the "coerces to logical without becoming NA" test, which seems like something you want to protect against. So how about adding a case like

```
x_numbers <- c(-1, 0, 0.1, 1, 2)  # should not be detected as vector of logicals
```

That said, if the tests for `util.is_vector_of_numbers` already cover these cases, no need to repeat them here.